node/src/components/storage.rs

```diff
@@ -2042,6 +2013,52 @@ impl Storage {
             .expect("Could not commit transaction");
         switch_block
     }
+
+    /// Retrieves a single block header by height by looking it up in the index and returning it.
+    pub fn read_block_header_by_height(
+        &self,
+        height: u64,
+    ) -> Result<Option<BlockHeader>, FatalStorageError> {
+        let mut tx = self.storage.env.begin_ro_txn()?;
+
+        self.storage
+            .block_height_index
+            .get(&height)
+            .and_then(|block_hash| {
+                self.storage
+                    .get_single_block_header(&mut tx, block_hash)
+                    .transpose()
+            })
+            .transpose()
+    }
+
+    /// Retrieves a single block by height by looking it up in the index and returning it.
+    fn get_block_by_height(&self, height: u64) -> Result<Option<Block>, FatalStorageError> {
+        let mut tx = self.storage.env.begin_ro_txn()?;
+
+        self.storage.get_block_by_height(&mut tx, height)
+    }
```

Comment:
It's a bit weird that these two methods are so simliar yet so different:
1. First one is called _read_ block header, while the second one _get_ block
2. The first one is `pub`, the second is private.
3. The first one has to implement the logic itself, the second one delegates to `self.storage` 
:)

I'm surprised you didn't make it consistent.

Reply:
It did the minimum of changes to implement the parallel feature and kept all inconsistencies and such "intact". This was to minimize potential issues when/if this has to be ported to other branches.

I expect a more thorough cleanup of storage to come later with the 2.0 push.